components/builder-scheduler/src/server/scheduler.rs

```diff
@@ -437,6 +463,58 @@ impl ScheduleMgr {
         Ok(())
     }
 
+    fn promote_package(&self, ident: &OriginPackageIdent, channel: &str) -> Result<()> {
+        debug!("Promoting '{:?}' to '{}'", ident, channel);
+
+        // We re-create the depot client instead of caching and re-using it due to the
+        // connection getting dropped when it is attempted to be re-used. This _may_ be
+        // a known issue with hyper connection pool getting reset in some cases.
+        let depot_cli = depot_client::Client::new(&self.depot_url, PRODUCT, VERSION, None).unwrap();
```

Comment:
It might be worth making a note to revisit this code when we upgrade hyper to 0.11.x, which uses Tokio under the hood, and will require a fair amount of changes to our http-client.